### DISCUSSION TITLE
Is it possible to use gpt-2 models with llama.cpp?

### DISCUSSION THREAD

**Discussion by guinmoon** [ggml-org/llama.cpp#4380] [Q&A]
Hi. I want to say thanks for the very cool projects llama.cpp and ggml. I use them in [my iOS app](https://github.com/guinmoon/LLMFarm). I built some of the core of my project from llama.cpp, some from examples like gpt-2.
In the llama.cpp code I found these lines:

```c++
enum llm_arch {
    LLM_ARCH_LLAMA,
    LLM_ARCH_FALCON,
    LLM_ARCH_BAICHUAN,
    LLM_ARCH_GPT2,
    LLM_ARCH_GPTJ,
    LLM_ARCH_GPTNEOX,
    LLM_ARCH_MPT,
    LLM_ARCH_STARCODER,
    LLM_ARCH_PERSIMMON,
    LLM_ARCH_REFACT,
    LLM_ARCH_BLOOM,
    LLM_ARCH_STABLELM,
    LLM_ARCH_QWEN,
    LLM_ARCH_UNKNOWN,
};
```

And I can assume that somehow llama.cpp supports gpt-2 models. 

Am I right?

**Comment by cmp-nct** [comment]
I don't know if it's fully supported, you'll have to try.
There is a GPT-2 demo available, so that's a good sign at least : https://github.com/ggerganov/ggml/tree/master/examples/gpt-2

**Reply by guinmoon** [reply]
Thank you. I'm already using the code from this example in my application :). Unfortunately mmap is not implemented in the example.
